### PR TITLE
Rebase of errors-are-fine

### PR DESCRIPTION
**Pull request by gbaraldi**
I messed up when merging, Probably because the commits were the same, so the revertsa applied.
This looks fine now, but there might be something wrong

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
+
+
+# local method table for device functions
+@static if isdefined(Base.Experimental, Symbol("@overlay"))
```
Comment (giordano) [inline src/utils.jl:4]: Top-level `@static` is useless (and I think you're missing the `end`?)

Comment (giordano) [reply src/utils.jl:4]: No, the `end` is fine, it's just 4 lines below, but indentation of the whole file is off, which confused me

Comment (gbaraldi) [reply src/utils.jl:4]: Those `@static`s were present in CUDA.jl so I just copied them

**Comment by brenhinkeller** [comment]
Ok, so now we just need to figure out what in here is causing the dynamic dispatch / `_ijl_apply_generic` & co..

**Comment by brenhinkeller** [comment]
Ooh, integration tests pass on Nightly now, that's new!

**Comment by tshort** [comment]
@gbaraldi, I'm curious why you closed this. Given that it looks like it works on Julia v1.9, this'd be a nice feature to have.

**Comment by gbaraldi** [comment]
@tshort I forgot to add a comment, I closed it because I started this PR from zero and it's already merged here https://github.com/tshort/StaticCompiler.jl/pull/97

**Comment by tshort** [comment]
Great news! Thanks for the update. I see it now. This will be fun to play with.

**Comment by brenhinkeller** [comment]
Oh yeah, superseded by #97

**Pull request closed without merging**